weston-ivi-shell/src/ivi-controller.c

```diff
@@ -1889,13 +1905,18 @@ surface_event_configure(struct wl_listener *listener, void *data)
     uint32_t surface_id;
     struct weston_surface *w_surface;
 
+    w_surface = lyt->surface_get_weston_surface(layout_surface);
+    /* ivi_shell only emit the surface configured event when
+       it have the content and new size of buffer. Otherwise,
+       when the id-agen set the surface id for surface. */
+    if (!weston_surface_has_content(w_surface))
```

Comment:
Why do we need this specific condition now?

Reply:
I see when id-agent sets surface id then also configure event is emitted. its clear, this condition is needed